### PR TITLE
Update wash build with build_command

### PR DESCRIPTION
**Pull request by ericgregory**
Adding build_command instructions to wash build page.

### CODE REVIEW

File: docs/cli/build.md [hunk 1]
```diff
@@ -5,14 +5,34 @@ sidebar_position: 2
 description: "wash build command reference"
 ---
 
-To build custom wasmCloud entities like actors, providers or interfaces, `wash build` comes in as a handy tool. One can bring in their own project containing some business logic and build these entities by simply providing the path to the project or its associated `wasmcloud.toml` file. The built artifact is signed using automatically generated keys or a user may use their own keys to sign the build.
+`wash build` enables users to build and sign custom wasmCloud entities like actors, providers, or interfaces. One can bring in their own project containing some business logic and build these entities by simply providing the path to the project or its associated `wasmcloud.toml` file. The built artifact is signed using automatically generated keys or a user may use their own keys to sign the build.
```
Comment (lachieh) [inline docs/cli/build.md]:
```suggestion
`wash build` enables you to build and sign custom wasmCloud entities like actors, providers, or interfaces. You can bring in your own project containing some business logic and build these entities by simply providing the path to the project or its associated `wasmcloud.toml` file. The built artifact is signed using automatically generated keys or you may use your own keys to sign the build.
```

File: docs/cli/build.md [hunk 2]
```diff
@@ -5,14 +5,34 @@ sidebar_position: 2
...
+type = "actor"
+version = "0.1.0"
+
+[actor]
+claims = ["wasmcloud:httpclient", "wasmcloud:httpserver"]
+build_command = "componentize-py -d ../../wit -w wasi:http/proxy@0.2.0-rc-2023-12-05 componentize app -o http.wasm"
+build_artifact = "http.wasm"
+```
+
+When using the `build_command` field, users must also specify the `build_artifact`: the filename and location of the Wasm artifact that will be produced by the build command. This provides `wash build` with a target for signing the artifact.
```
Comment (lachieh) [inline docs/cli/build.md]:
```suggestion
If you use the `build_command` field, you must also specify the `build_artifact`: the filename and location of the Wasm artifact that will be produced by the build command. This provides `wash build` with a target for signing the artifact.
```

**Review by lachieh — APPROVED** [review]
I know this language existed before, but prefer talking direct to people in second person than third-person. Feel free to disagree and commit, since I'm no wordperson.

**Comment by ericgregory** [comment]
@lachieh 100% agree. We'll want to adopt an approach and apply it throughout, but I prefer the second person, too.

**Review by lachieh — APPROVED** [review]

**Review by lachieh — APPROVED** [review]
I think that because I helped with those recent commits, it's not going to let me approve this. @connorsmith256 might have to grab this one.

It looks good to me so as long as it builds then, @ericgregory, you're free to take it out of draft and merge it.

File: docs/cli/build.md [hunk 3]
```diff
@@ -5,14 +5,34 @@ sidebar_position: 2
...
+
+[actor]
+claims = ["wasmcloud:httpclient", "wasmcloud:httpserver"]
+build_command = "componentize-py -d ../../wit -w wasi:http/proxy@0.2.0-rc-2023-12-05 componentize app -o http.wasm"
+build_artifact = "http.wasm"
+```
+
+If you use the `build_command` field, you must also specify the `build_artifact`: the filename and location of the Wasm artifact that will be produced by the build command. This provides `wash build` with a target for signing the artifact.
+
+**Note**: Currently the `language` field in the `wasmcloud.toml` metadata must be set to an "officially supported" language (i.e. Rust or Go) to avoid errors. The `build_command` field does not have full support for environment variables or multiple commands and should be in the form of "command arg1 arg2 arg...". Use an external script to handle more complex build commands.
```
Comment (vados-cosmonic) [inline docs/cli/build.md]: NIT: we do have a callout syntax (`::info`? something like that), but can be fixed in a follow up!

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket:

File: docs/cli/build.md (hunk 3 above)
Comment (lachieh) [reply docs/cli/build.md]: @vados-cosmonic, good point.

@ericgregory These are called [admonitions](https://docusaurus.io/docs/markdown-features/admonitions) and they work like this:
https://github.com/wasmCloud/wasmcloud.com/blob/d6e9e1bfb3932e829293e817d19a6a64f84638ed/docs/ecosystem/wadm/index.md?plain=1#L16-L18

Comment (ericgregory) [reply docs/cli/build.md]: @vados-cosmonic @lachieh Ah, that syntax will definitely be helpful—thanks both!

**Review by lachieh — APPROVED** [review]

**Review by connorsmith256 — APPROVED** [review]
🚀

**Pull request merged**